### PR TITLE
RHEL8 FreeIPA on AWS and AWS_GOV

### PR DESCRIPTION
**Pull request by pkedvessy**

### CODE REVIEW

File: saltstack/base/salt/prerequisites/jinja.sls [hunk 1]
```diff
@@ -1,3 +1,9 @@
+{% if pillar['OS'] == 'redhat8' %}  
 install_jinja2:
   cmd.run:
```
Comment (TheTinkerDad) [inline saltstack/base/salt/prerequisites/jinja.sls:3]: You can use this instead:

```
install_jinja2:
  pip.installed:
      - name: jinja2
```

pip3 is not always available or on the path - the above solution takes the pip available for Salt and uses that one - usually the one that is bound to the virtualenv Salt is running in. With that, you don't need the "if" to differentiate between OSes.

File: saltstack/base/salt/prerequisites/pip.sls [hunk 2]
```diff
@@ -24,15 +24,21 @@ update_python_pip2:
...
-    - onlyif: pip3 -V
+    - name: python3 -m pip install --upgrade --index=https://pypi.python.org/simple/ pip==8.1.2
+    - onlyif: python3 -m pip -V
 
 {% endif %}
 
+{% if pillar['OS'] == 'redhat8' %}  
+install_pyyaml:
+  cmd.run:
+    - name: python3 -m pip install "PyYAML>=5.1" --ignore-installed
```
Comment (TheTinkerDad) [inline saltstack/base/salt/prerequisites/pip.sls:35]: Suggestion:

```
install_pyyaml:
  pip.installed:
    - name: PyYAML
```
    
Also, we're installing PyYAML 6.0 by default either way - you can get rid of the whole "if" part too

File: saltstack/freeipa/salt/ipaconsistency/init.sls [hunk 3]
```diff
@@ -1,6 +1,6 @@
 install_checkipaconsistency:
   cmd.run:
-    - name: pip install checkipaconsistency==2.7.10
+    - name: python3 -m pip install checkipaconsistency==2.7.10
```
Comment (TheTinkerDad) [inline saltstack/freeipa/salt/ipaconsistency/init.sls:3]: Suggestion:

```
install_checkipaconsistency:
  pip.installed:
    - name: checkipaconsistency==2.7.10
```

File: scripts/generate-delta-package-list.sh [hunk 4]
```diff
@@ -73,7 +73,11 @@ function collect_rpm_packages() {
 function collect_python_packages() {
   file=$1
   echo "Collecting installed python packages"
-  pip list --format json | jq -r '.[].name' | sort >> "$file"
+  if [ "${OS}" == "redhat8" ] ; then
```
Comment (TheTinkerDad) [inline scripts/generate-delta-package-list.sh:76]: You could replace "virtualenv" and "pip" with "python3 -m virtualenv" and "python3 -m pip" in general - we shouldn't rely on anything Python 2 related nowadays, not even on CentOS 7.

File: scripts/get-salt-version.sh [hunk 5]
```diff
@@ -30,7 +30,7 @@ compare_version () {
 BASE_NAME=$1
 STACK_VERSION=$2
 
-SALT_VERSION=3001.8
+SALT_VERSION=3004.1
```
Comment (TheTinkerDad) [inline scripts/get-salt-version.sh:33]: Isn't 3004.1 something we previously couldn't update to? Or is this an RHEL / Gov related mandatory change? Personally I wouldn't do this together with all of the changes required for Gov, but instead separately under https://jira.cloudera.com/browse/CB-16854 on master and test it with CentOS 7 first.

File: scripts/salt-install.sh [hunk 6]
```diff
@@ -20,14 +20,30 @@ function install_salt_with_pip() {
...
     pip install pbr
   fi
   pip install --upgrade pip
   pip install -r /tmp/salt_requirements.txt
 }
 
+function install_salt_with_pip3() {
+
+  echo "Installing salt with version: $SALT_VERSION"
+  pip3 install --upgrade pip
```
Comment (TheTinkerDad) [inline scripts/salt-install.sh:34]: At some places you've removed "pip3" and replaced it with "python3 -m pip". Why do you stick to "pip3" here?

**Review by TheTinkerDad — COMMENTED** [review]
I'm not requesting changes, only suggested a few - I'll leave it up to you whether you want to adopt them or not. Otherwise, LGTM

**Review by TheTinkerDad — APPROVED** [review]

File: scripts/get-salt-version.sh (hunk 5 above)
Comment (pkedvessy) [reply scripts/get-salt-version.sh:33]: Agree, it came in dunno why... Maybe it was there by a mistake on the base branch I used.

File: saltstack/base/salt/prerequisites/jinja.sls (hunk 1 above)
Comment (TheTinkerDad) [reply saltstack/base/salt/prerequisites/jinja.sls:3]: Already done on another branch

File: saltstack/freeipa/salt/ipaconsistency/init.sls (hunk 3 above)
Comment (TheTinkerDad) [reply saltstack/freeipa/salt/ipaconsistency/init.sls:3]: Already done on another branch

**Pull request closed without merging**